### PR TITLE
Add README's for packages/forms

### LINKED ISSUE #30503: Jetpack Forms - Create readme file for new Form Responses inbox
For the new “Form Responses” inbox view, let's add [here](https://github.com/Automattic/jetpack/tree/trunk/projects/packages/forms/src/dashboard) some readme file … similar to [one of these](https://github.com/Automattic/wp-calypso/tree/trunk/apps/odyssey-stats), so that people would have an easy start to update things.

### PR DESCRIPTION
**Pull request by ice9js**
Fixes #30503.

## Proposed changes:

This PR adds README files for the `forms` package in an effort to hopefully make it more approachable for future maintainers or users.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

N/A

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

The PR doesn't contain any changes to the code. No testing necessary.

### CODE REVIEW

**Review by CGastrell — DISMISSED** [review]
LGTM! :shipit:

File: projects/packages/forms/README.md [hunk 1]
```diff
@@ -1,14 +1,106 @@
-# forms
+# Jetpack Forms
 
-Jetpack Forms
+All things forms. This package currently contains them implementation for Jetpack's Contact Form module, Contact Form Gutenberg blocks and the new Jetpack Forms feedback dashboard.
```
Comment (jeherve) [inline projects/packages/forms/README.md]:
```suggestion
All things forms. This package currently contains the implementation for Jetpack's Contact Form module, Form blocks, and the new Jetpack Forms feedback dashboard.
```

File: projects/packages/forms/README.md [hunk 2]
```diff
@@ -1,14 +1,106 @@
...
 
-## How to install forms
+## Hierarchy
 
-### Installation From Git Repo
+```
+.
+├── src/
+│   ├── blocks/                                      - Contact form Gutenberg blocks.
```
Comment (jeherve) [inline projects/packages/forms/README.md]:
```suggestion
│   ├── blocks/                                      - Form blocks.
```

**Review by CGastrell — APPROVED** [review]
Still looking good, nice suggestions from Jeremy

**Pull request merged**